### PR TITLE
Fix weird behavior of "Select all" checkbox

### LINKED ISSUE #301: "Select all" checkbox behavior
I think that behavior of "Select all" checkbox (in `/moments/new` and `/strategies/new`) is kinda weird

image

It should be rather something like this http://codepen.io/veerkun/pen/oAHCG

What do you think about it?

### PR DESCRIPTION
**Pull request by Wowu**
- Now checkbox will be checked even if user clicks on the checkbox label
- Fix #301 
- Add new contributor to pages_controller :tada:

### CODE REVIEW

**Comment by Wowu** [comment]
I'll update tests tommorow.

File: app/views/strategies/_form.html.erb [hunk 1]
```diff
@@ -58,10 +58,16 @@
...
+      <label>
+        <input type="checkbox" id="viewers_all" />
+        <span id="viewers_label"><%= t('select_all') %></span>
+      </label>
       <br>
       <% @viewers.each do |item| %>
-      <%= f.check_box(:viewers, {:multiple => true, :checked => @strategy.viewers.include?(item.id)}, item.id, nil) %><%= User.where(:id => item.id).first.name %><br>
+        <label>
+          <%= f.check_box(:viewers, {:multiple => true, :checked => @strategy.viewers.include?(item.id)}, item.id, nil) %>
+          <%= User.where(:id => item.id).first.name %><br>
```
Comment (julianguyen) [inline app/views/strategies/_form.html.erb]: Extra linebreak issue here too

File: config/locales/en.yml [hunk 2]
```diff
@@ -22,6 +22,7 @@ en:
  strategy: 'Strategy'
  add: 'Add'
  add_all: 'Add all'
+ select_all: 'Select all'
```
Comment (julianguyen) [inline config/locales/en.yml]: Nice 💯

**Comment by zachariast** [comment]
@Wowu do you update the tests? because i am getting some errors, and it says about tests.

**Comment by Wowu** [comment]
@zachariast, what do you mean exactly, this pull request?

**Comment by zachariast** [comment]
@Wowu in my pull request about clear button i got some errros about tests. And yesterday you said that you will update some test, and i thought that those tests are the problem to my pull request. Am i wrong?

**Comment by Wowu** [comment]
I wrote it here, in my pull request, so it was related to my work, but I can take a look at your issue too :smiley: 
Remember that discussion here should be only about this pull request. If you have any notices or questions about your work you should leave them in #304. Anyway, I'll check your code soon.

**Comment by julianguyen** [comment]
Tests LGTM!

**Comment by julianguyen** [comment]
@zachariast @Wowu, ideally we would like the Codeclimate checks to pass too, but we can sometimes ignore what is suggested. We definitely want the Travis CI checks to pass though!

This is up to you @wowu, but you could address the first two suggestions in https://codeclimate.com/github/julianguyen/ifme/pull/302, perhaps by moving the contents of the click event into a shared helper in application.js. It's up to you though. Feel free to merge when you're ready!

**Comment by Wowu** [comment]
@julianguyen, can you check if my tests are written correctly? If me is my first project where I'm dealing with tests actually.

File: spec/javascripts/MomentsSpec.js [hunk 3]
```diff
@@ -1,105 +1,36 @@
...
-
-      expect(setViewersCheckBoxToNotBeSelected).toHaveBeenCalled();
+  afterAll(function() {
+    $("#viewers_list").remove();
   });
 
-  it("has set #viewers_label text to NO_ALLIES when ALL_ALLIES is selected", function() {
-      isAllAlliesInputBoxIsChecked.and.returnValue(true);
+  it("has selected all allies who can view moment when \"Select all\" is selected",  function() {
+    $('#viewers_all').click();
```
Comment (julianguyen) [inline spec/javascripts/MomentsSpec.js]: You can also test `expect($(":checkbox[name='moment[viewers][]']").eq(0).prop("checked")).toBe(false);` before clicking

File: spec/javascripts/StrategiesSpec.js [hunk 4]
```diff
@@ -24,33 +24,31 @@ describe("Strategies", function() {
...
     newOrEdit.and.returnValue(true);
 
-    onReadyStrategies();
+    onReadyMomentsAndStrategies();
 
-    expect($(":checkbox[name='strategy[viewers][]']").prop("checked")).toBe(false);
-    expect($(":checkbox[id='viewers_all']").prop("checked")).toBe(false);
-  	expect($('#viewers_label').text()).toBe(ALL_ALLIES);
-  });
+    $('#viewers_all').click();
```
Comment (julianguyen) [inline spec/javascripts/StrategiesSpec.js]: You can also test ` expect($(":checkbox[name='strategy[viewers][]']").eq(0).prop("checked")).toBe(false);` before the click.

**Comment by julianguyen** [comment]
@Wowu You're tests look fine to me. I add a few comments too!

**Pull request merged**